version/version.go

```diff
@@ -8,18 +8,22 @@
 package version
 
 import (
+	_ "embed"
 	"fmt"
-
 	version "github.com/hashicorp/go-version"
+	"strings"
 )
 
-// The main version number that is being run at the moment.
-var Version = "1.5.0"
+var (
+	//go:embed VERSION
+	fullVersion string
 
-// A pre-release marker for the version. If this is "" (empty string)
-// then it means that it is a final release. Otherwise, this is a pre-release
-// such as "dev" (in development), "beta", "rc1", etc.
-var Prerelease = "dev"
+	// The main version number that is being run at the moment.
+	// A pre-release marker for the version. If this is "" (empty string)
+	// then it means that it is a final release. Otherwise, this is a pre-release
+	// such as "dev" (in development), "beta", "rc1", etc.
+	Version, Prerelease, _ = strings.Cut(strings.TrimSpace(fullVersion), "-")
```

Comment:
I've got an [existing PR](https://github.com/hashicorp/terraform/pull/32973) along these lines that I was waiting to merge until after we cut the 1.5 release branch! One thing that came up when discussing that PR was that we do find value in having the `dev` prerelease marker in local development and would be sad to give it up, so that PR includes a switch to let us keep `dev` in local development while reading the version directly from `VERSION` for release builds.

Reply:
Yes this makes a lot of sense to me. I'm going to remove the version bits from this PR and link to yours in the description. The other workflows in this PR can then just take care of changelog specific things 👍

Reply:
Sounds good!